src/components/FiatOnrampModal/getDefaultCurrencyCode.ts

```diff
@@ -0,0 +1,76 @@
+import { ChainId, WETH9 } from '@uniswap/sdk-core'
+import {
+  MATIC,
+  USDC_ARBITRUM,
+  USDC_MAINNET,
+  USDC_OPTIMISM,
+  USDC_POLYGON,
+  USDT,
+  WBTC,
+  WETH_POLYGON,
+} from 'constants/tokens'
+import { Chain } from 'graphql/data/__generated__/types-and-hooks'
+import { validateUrlChainParam } from 'graphql/data/util'
+
+import { MoonpaySupportedCurrencyCode } from './constants'
+
+const CURRENCY_CODES: {
+  [K in Chain]: {
+    [key: string]: MoonpaySupportedCurrencyCode
+    NATIVE: MoonpaySupportedCurrencyCode
+  }
+} = {
+  [Chain.Ethereum]: {
+    [WETH9[ChainId.MAINNET]?.address]: 'weth',
+    [USDC_MAINNET.address]: 'usdc',
+    [USDT.address]: 'usdt',
+    [WBTC.address]: 'wbtc',
+    [MATIC.address]: 'polygon',
+    NATIVE: 'eth',
+  },
+  [Chain.Arbitrum]: {
+    [USDC_ARBITRUM.address]: 'usdc_arbitrum',
+    NATIVE: 'eth_arbitrum',
+  },
+  [Chain.Optimism]: {
+    [USDC_OPTIMISM.address]: 'usdc_optimism',
+    NATIVE: 'eth_optimism',
+  },
+  [Chain.Polygon]: {
+    [USDC_POLYGON.address]: 'usdc_polygon',
+    [WETH_POLYGON.address]: 'eth_polygon',
+    NATIVE: 'matic_polygon',
+  },
+  // Defaults for unsupported chains, needed to satisfy type checker.
```

Comment:
to work around this you can just define a type for the subset of Chain for only the Moonpay supported ones